### PR TITLE
src/rgw: Dpp addition to create bucket log (Resolved Conflicts)

### PR DESCRIPTION
**Pull request by TRYTOBE8TME**
Signed-off-by: Kalpesh Pandya <kapandya@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/rgw/librgw.cc [hunk 1]
```diff
@@ -242,7 +242,7 @@ namespace rgw {
     /* XXX and -then- stash req_state pointers everywhere they are needed */
     ret = req->init(rgw_env, &rados_ctx, io, s);
     if (ret < 0) {
-      dout(10) << "failed to initialize request" << dendl;
+      ldpp_dout(s, 10) << "failed to initialize request" << dendl;
```
Comment (alimaredia) [inline src/rgw/librgw.cc:245]: @TRYTOBE8TME change the `s` to `op` here

File: src/rgw/librgw.cc [hunk 2]
```diff
@@ -302,9 +302,9 @@ namespace rgw {
       ret = op->verify_permission(null_yield);
       if (ret < 0) {
 	if (s->system_request) {
-	  dout(2) << "overriding permissions due to system operation" << dendl;
+	  ldpp_dout(s, 2) << "overriding permissions due to system operation" << dendl;
```
Comment (alimaredia) [inline src/rgw/librgw.cc:305]: Ditto

File: src/rgw/librgw.cc [hunk 3]
```diff
@@ -302,9 +302,9 @@ namespace rgw {
...
 	} else if (s->auth.identity->is_admin_of(s->user->get_id())) {
-	  dout(2) << "overriding permissions due to admin operation" << dendl;
+	  ldpp_dout(s, 2) << "overriding permissions due to admin operation" << dendl;
```
Comment (alimaredia) [inline src/rgw/librgw.cc:307]: Ditto

File: src/rgw/librgw.cc [hunk 4]
```diff
@@ -344,7 +344,7 @@ namespace rgw {
 
     ldpp_dout(s, 2) << "http status=" << http_ret << dendl;
 
-    dout(1) << "====== " << __func__
+    ldpp_dout(s, 1) << "====== " << __func__
```
Comment (alimaredia) [inline src/rgw/librgw.cc:347]: Ditto

File: src/rgw/librgw.cc [hunk 5]
```diff
@@ -380,7 +380,7 @@ namespace rgw {
 
     int ret = req->init(rgw_env, &rados_ctx, &io_ctx, s);
     if (ret < 0) {
-      dout(10) << "failed to initialize request" << dendl;
+      ldpp_dout(s, 10) << "failed to initialize request" << dendl;
```
Comment (alimaredia) [inline src/rgw/librgw.cc:383]: Ditto

File: src/rgw/librgw.cc [hunk 6]
```diff
@@ -434,9 +434,9 @@ namespace rgw {
     ret = op->verify_permission(null_yield);
     if (ret < 0) {
       if (s->system_request) {
-	dout(2) << "overriding permissions due to system operation" << dendl;
+	ldpp_dout(s, 2) << "overriding permissions due to system operation" << dendl;
```
Comment (alimaredia) [inline src/rgw/librgw.cc:437]: Ditto

File: src/rgw/librgw.cc [hunk 7]
```diff
@@ -434,9 +434,9 @@ namespace rgw {
...
       } else if (s->auth.identity->is_admin_of(s->user->get_id())) {
-	dout(2) << "overriding permissions due to admin operation" << dendl;
+	ldpp_dout(s, 2) << "overriding permissions due to admin operation" << dendl;
```
Comment (alimaredia) [inline src/rgw/librgw.cc:439]: Ditto

**Comment by alimaredia** [comment]
@TRYTOBE8TME in `rgw_file.cc` you can remove the `DoutPrefixProvider` in `RGWWriteRequest::exec_finish` since it's not necessary. To do so you'll also have to change the call to `exec_finish` in `RGWLibContinuedReq` in `rgw_lib.h` since `RGWWriteRequest` inherits from  `RGWLibContinuedReq`.

**Comment by TRYTOBE8TME** [comment]
http://pulpito.front.sepia.ceph.com/kapandya-2021-02-04_08:02:46-rgw-master-distro-basic-smithi/

**Comment by alimaredia** [comment]
This PR is ready to merge, I'll approve once you change the commit message. Could you add a body to the commit message and change `src/rgw:` to `rgw:` and remove `(Resolved Conflicts)` in the subject line.

**Review by alimaredia — APPROVED** [review]

**Comment by tchaikov** [comment]
@alimaredia could you add a Reviewed-by line(s) in the commit message of merge commit when merging PRs in future?

**Comment by alimaredia** [comment]
@tchaikov yes, that's my fault. For some reason I thought github automatically did that for us.

**Pull request merged**